### PR TITLE
Fix/improve 'must free heap allocations' code

### LINKED ISSUE #9183: The LRUCache::DisownData() drops data in case code is compiled with ASAN
It looks like MUST_FREE_HEAP_ALLOCATIONS is not properly defined in scope of `cache/lru_cache.cc`. As a result ASAN leak warnings suppression doesn't work as expected. The issue seem to be caused by changes in https://github.com/facebook/rocksdb/commit/311a544c2aa513a1f9b33823996f6b3e7843b6c5.

The `ClockCache::DisownData()` should be affected as well.

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: Added missing include, and cleaned up to make same mistake less
likely in future (minimize conditional compilation)

Fixes #9183

Test Plan: added to existing test

### CODE REVIEW

File: port/lang.h [hunk 1]
```diff
@@ -42,8 +42,10 @@
 //   STATIC_AVOID_DESTRUCTION(Foo, foo)(arg1, arg2);
 #ifdef MUST_FREE_HEAP_ALLOCATIONS
 #define STATIC_AVOID_DESTRUCTION(Type, name) static Type name
+constexpr bool kMustFreeHeapAllocations = true;
 #else
 #define STATIC_AVOID_DESTRUCTION(Type, name) static Type& name = *new Type
+constexpr bool kMustFreeHeapAllocations = true;
```
Comment (mrambacher) [inline port/lang.h:48]: Shouldn't this one be false?

Comment (pdillinger) [reply port/lang.h:48]: Quite a copy-paste error there! Oops

**Review by mrambacher — APPROVED** [review]
Fix the copy-and-paste, and the rest looks good.

**Review by mrambacher — APPROVED** [review]
LGTM.

**Pull request closed without merging**